assets/src/edit-story/elements/gif/output.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import StoryPropTypes from '../../types';
+import MediaOutput from '../media/output';
+
+function defaultForUndefined(value, def) {
+  return value === undefined ? def : value;
+}
+
+function GifOutput({ element, box }) {
+  const { resource } = element;
+
+  const sourceProps = {
+    type: resource.output.mimeType,
+    src: resource.output.src,
+  };
+
+  const props = {
+    autoPlay: 'autoplay',
+    poster: '',
+    title: defaultForUndefined(element.title, resource.title),
+    alt: defaultForUndefined(element.alt, resource.alt),
+    layout: 'fill',
+    loop: 'loop',
+    noaudio: 'noaudio',
+  };
```

Comment:
I'd rather just defined these props inline in the returned JSX. This style (defining prop objects and spreading in JSX) is mainly used for variable prop lists, not static ones like this.

Then the boolean ones also become more compact (just `<amp-video autoPlay />` rather than `<amp-video autoPlay="autoplay" />`).

Reply:
note: `amp-video` gave me errors with booleans for loop, autoPlay and noaudio

Reply:
Interesting - which errors? And where did they occur?

Reply:
Maybe AMP is weird regarding boolean props and requires values. If that's the case, that's how we do it of course.

Reply:
Perhaps error is the wrong word... it was functional, but the unit test fails at `expect(outputStr).toBeValidAMPStoryElement()` with the definition/error below, suggesting the boolean values are invalid (I'll keep the string values for these for now, that pattern is in `video/output` for the loop attribute as well, maybe that's something to fix in the test matchers)

```javascript
    <MediaOutput element={element} box={box}>
      <amp-video
        id={`el-${element.id}-media`}
        autoplay
        loop
        noaudio
        poster=""
        layout="fill"
        title={element.title ?? resource.title}
        alt={element.alt ?? resource.alt}
      >
        <source type={resource.output.mimeType} src={resource.output.src} />
      </amp-video>
    </MediaOutput>
```

```bash
> jest --config=tests/js/jest.config.js "./assets/src/edit-story/elements/gif/test/output.js"

.
  ● Gif Output › should produce valid AMP output

    Expected <div style="position:absolute;width:100%;height:100%;left:0%;top:0%"><amp-video id="el-048426c8-69ae-4e04-80f1-8f3fd4434261-media" autoplay="true" loop="true" noaudio="true" poster="" layout="fill" alt="" crossorigin="anonymous"><source type="video/mp4" src="https://c.tenor.com/4F2m7BWP6KYAAAPo/flying-kiss-muah.mp4"/></amp-video></div> to be valid AMP. Errors:
    The attribute 'autoplay' in tag 'amp-video' is set to the invalid value 'true'. (see https://amp.dev/documentation/components/amp-video/)
    The attribute 'loop' in tag 'amp-video' is set to the invalid value 'true'. (see https://amp.dev/documentation/components/amp-video/)
    The attribute 'noaudio' in tag 'amp-video' is set to the invalid value 'true'. (see https://amp.dev/documentation/components/amp-video/)

      66 |   };
      67 |   it('should produce valid AMP output', async () => {
    > 68 |     await expect(<GifOutput {...baseProps} />).toBeValidAMPStoryElement();
         |                                                ^
      69 |   });
      70 |   it('should produce an AMP video with autoplay, no controls, no audio, and loop', async () => {
      71 |     const output = <GifOutput {...baseProps} />;

      at Object.<anonymous> (assets/src/edit-story/elements/gif/test/output.js:68:48)
```

however... the [example for autoplay in the amp.dev docs](https://amp.dev/documentation/examples/components/amp-video/#autoplay) uses a boolean...

Reply:
If you just use boolean attributes, on output React will turn them into `loop="true"` instead of the required `loop="loop"`

That's actually why we added these AMP validation tests in the first place, to prevent exactly this from happening again :-)

So yeah, the current way is fine.